### ISSUE TITLE
BreezeWiki host menu is empty

### ISSUE THREAD

**Issue by Duckrinium** [KevinPayravi/indie-wiki-buddy#618]
Firefox 124.0.1 (Flatpak if it matters)

The title is pretty self-explanatory. I've noticed this long time ago but only now decided to check the console output.

Console says next:
- `XHRGET
https://bw.getindie.wiki/instances.json
[HTTP/2 403  55ms]`
- `Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://bw.getindie.wiki/instances.json. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing). Status code: 403.`
- `Indie Wiki Buddy failed to get BreezeWiki data: TypeError: NetworkError when attempting to fetch resource. common-page-functions.js:303:19`

And here what it looks like:
![image](https://github.com/KevinPayravi/indie-wiki-buddy/assets/37019004/0d8a70b0-36ba-47fd-a3ea-83896211b988)

Seems to be caused by a privacy measure from Firefox.

**Comment by KevinPayravi** [comment]
@Duckrinium Thanks for reporting this!

I double-checked `https://bw.getindie.wiki/instances.json` and it is responding with the required `access-control-allow-origin: *` header. I know Firefox can return that CORS error for a whole range of issues that aren't actually due to CORS.

I also use Firefox and haven't seen this issue myself + this is the first report I've received on it, so I'm guessing it's something specific to your Firefox setup. Are you using any other extensions that might be blocking network requests? Ad blockers, Privacy Badger, etc.?

**Comment by Duckrinium** [comment]
> Are you using any other extensions that might be blocking network requests?

Yes, uBlock Origin and DuckDuckGo Essentials, and I assumed they don't touch other extensions due to constrains on what extensions can do to each other. I can see all of them say "Can't access..." in extensions menu on settings page.

However, disabling them does not seem to affect anything. Perhaps I need to change something in my browser settings?

**Comment by KevinPayravi** [comment]
Not sure! I'll continue looking into it and how we might be able to replicate.

On another note, I also created a new task (https://github.com/KevinPayravi/indie-wiki-buddy/issues/621) to load a default list if that initial call to fetch BreezeWiki mirrors fails.

**Comment by Duckrinium** [comment]
Also tried running:
- firefox from nixos repo
- from windows 10 vm
None worked.

**Comment by Ilyasnow** [comment]
I'm also experiencing the same issue on Google Chrome (Version 123.0.6312.106) with the same CORS error:
> Access to fetch at 'https://bw.getindie.wiki/instances.json' from origin 'chrome-extension://fkagelmloambgokoeokbpihmgpkbgbfm' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

And despite @KevinPayravi affirming that the header should be present, that is not my experience. I'm attaching a header overview from Chrome web console:
[breezewiki_mirrors_log.txt](https://github.com/KevinPayravi/indie-wiki-buddy/files/14922409/breezewiki_mirrors_log.txt)

Also tried to check with Firefox (Version 124.0.1 (64-bit)) and the required header was not present either.

**Comment by KevinPayravi** [comment]
@Ilyasnow Thank you for sharing!

I've added a Cloudflare transform rule to add the `access-control-allow-origin: *` header to the response, which should hopefully result in the header always being attached. Please let me know if you're still experiencing the issue.

**Comment by Ilyasnow** [comment]
@KevinPayravi Thank you for the quick reply and fix!

It did indeed fix the `access-control-allow-origin` header issue. Unfortunately however, it now just returns a generic 403 (Forbidden) error on the same request. From what I could gather it seems to return a cloudflare captcha page on the same reply.

**Comment by KevinPayravi** [comment]
@Ilyasnow I just disabled the security features that I believe are causing that CAPTCHA - please give it another try!

**Comment by Ilyasnow** [comment]
@KevinPayravi Not sure that disabling captcha is a correct move generally. I believe most of the troubles related to the current issue can be resolved by the implementation of #621.

There also appears to be an issue with the response header:
> Access to fetch at 'https://bw.getindie.wiki/instances.json' from origin 'chrome-extension://fkagelmloambgokoeokbpihmgpkbgbfm' has been blocked by CORS policy: The 'Access-Control-Allow-Origin' header contains multiple values '*, *', but only one is allowed. Have the server send the header with a valid value, or, if an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

**Comment by KevinPayravi** [comment]
It's all just static content, so I'm not too concerned about disabling CAPTCHA. Cloudflare is also generally good at detecting abusive/bot traffic even without CAPTCHA. If needed, I can increase security again once #621 is implemented.

> There appear to be two headers now

Updated the settings so there should only be one now.

**Comment by Ilyasnow** [comment]
@KevinPayravi thank you! I can confirm that the list of hosts is displayed correctly for me now. 

Not sure if the OP had the same issue with the Cloudflare, however. @Duckrinium please check if your issue is resolved as well.

**Comment by Duckrinium** [comment]
> please check if your issue is resolved as well.
![image](https://github.com/KevinPayravi/indie-wiki-buddy/assets/37019004/2626774f-d630-409e-b6bc-78b1f3609740)
it works now, yay :D

**Comment by Duckrinium** [comment]
Android version also works so I'll be closing this issue as it is considered fixed. Thank you for an amazing extension!
